sdv/data_processing/data_processor.py

```diff
@@ -62,10 +80,217 @@ def _update_numerical_transformer(self, learn_rounding_scheme, enforce_min_max_v
         self._transformers_by_sdtype.update({'numerical': custom_float_formatter})
 
     def __init__(self, metadata, learn_rounding_scheme=True, enforce_min_max_values=True,
-                 model_kwargs=None):
+                 model_kwargs=None, table_name=None):
         self.metadata = metadata
         self._model_kwargs = model_kwargs or {}
         self._constraints = self._load_constraints()
         self._constraints_to_reverse = []
         self._transformers_by_sdtype = self._DEFAULT_TRANSFORMERS_BY_SDTYPE.copy()
         self._update_numerical_transformer(learn_rounding_scheme, enforce_min_max_values)
+        self._hyper_transformer = None
+        self.table_name = table_name
+        self._dtypes = None
+        self.fitted = False
+
+    def get_model_kwargs(self, model_name):
+        """Return the required model kwargs for the indicated model.
+
+        Args:
+            model_name (str):
+                Qualified Name of the model for which model kwargs
+                are needed.
+
+        Returns:
+            dict:
+                Keyword arguments to use on the indicated model.
+        """
+        return copy.deepcopy(self._model_kwargs.get(model_name))
+
+    def set_model_kwargs(self, model_name, model_kwargs):
+        """Set the model kwargs used for the indicated model."""
+        self._model_kwargs[model_name] = model_kwargs
+
+    def _fit_constraints(self, data):
+        errors = []
+        for constraint in self._constraints:
+            try:
+                constraint.fit(data)
+            except Exception as e:
+                errors.append(e)
+
+        if errors:
+            raise MultipleConstraintsErrors(errors)
+
+    def _transform_constraints(self, data, is_condition=False):
+        errors = []
+        if not is_condition:
+            self._constraints_to_reverse = []
+
+        for constraint in self._constraints:
+            try:
+                data = constraint.transform(data)
+                if not is_condition:
+                    self._constraints_to_reverse.append(constraint)
+
+            except (MissingConstraintColumnError, FunctionError) as e:
```

Comment:
I think it would be good if instead of `e` we used `error` for the future linting plugins.